### PR TITLE
layers: Add support VK_EXT_shader_tile_image 

### PR DESCRIPTION
**Pull request by mohfai-arm**
This PR include the missing barrier support for vkCmdPipelineBarrier when extension enabled.

### CODE REVIEW

File: layers/sync/sync_vuid_maps.cpp [hunk 1]
```diff
@@ -1168,4 +1168,27 @@ const std::string &GetQueueSubmitVUID(const Location &loc, SubmitError error) {
     return result;
 }
 
+const std::string &GetShaderTileImageVUID(const Location &loc, ShaderTileImageError error) {
```
[the remark this answers is not in the document]
Comment (juan-lunarg) [reply layers/sync/sync_vuid_maps.cpp:1171]: I agree but it's consistent with the other functions in this file. So it should remain `const std::string &` for this PR.

A separate PR would be best to cleanup the std::string / std::map usage.

[the remark this answers is not in the document]
Comment (juan-lunarg) [reply layers/sync/sync_vuid_maps.cpp:1171]: Best to cleanup after this PR gets merged in.

File: layers/core_checks/cc_synchronization.cpp [hunk 2]
```diff
@@ -2444,55 +2484,41 @@ bool CoreChecks::ValidateDependencyInfoForShaderTileImage(const Location &outer_
...
+    }
 
-        if (HasNonFramebufferStagePipelineStageFlags(mem_barrier.dstStageMask)) {
-            skip |= LogError(cb_state.commandBuffer(), "VUID-vkCmdPipelineBarrier-None-08719",
-                             "%s (%s) dstStageMask are restricted to framebuffer space stages (%s).",
-                             loc.dot(Field::dstStageMask).Message().c_str(),
-                             sync_utils::StringPipelineStageFlags(mem_barrier.dstStageMask).c_str(),
-                             sync_utils::StringPipelineStageFlags(kFramebufferStagePipelineStageFlags).c_str());
-        }
+    if constexpr (!std::is_same_v<Barrier, VkMemoryBarrier2>) {
```
[the remark this answers is not in the document]
Comment (mohfai-arm) [reply layers/core_checks/cc_synchronization.cpp]: done

File: layers/core_checks/cc_synchronization.cpp [hunk 3]
```diff
@@ -2433,9 +2444,38 @@ bool CoreChecks::ValidateDependencyInfo(const LogObjectList &objects, const Loca
...
+}
+
+template <typename Barrier>
+bool CoreChecks::ValidateBarriersForShaderTileImage(const LogObjectList &objlist, const Location &outer_loc,
+                                                    VkDependencyFlags dependencyFlags, uint32_t memBarrierCount,
+                                                    const Barrier *pMemBarriers, uint32_t bufferBarrierCount,
+                                                    uint32_t imageMemBarrierCount, VkPipelineStageFlags src_stage_mask,
+                                                    VkPipelineStageFlags dst_stage_mask) const {
+    bool skip = false;
+
```
[the remark this answers is not in the document]
Comment (mohfai-arm) [reply layers/core_checks/cc_synchronization.cpp:2476]: Done

**Review by arno-lunarg — APPROVED** [review]

**Comment by juan-lunarg** [comment]
Looks like Github Action server is still unhappy. I'm sure re-running the failed jobs will fix it.

**Review by juan-lunarg — APPROVED** [review]

**Review by ncesario-lunarg — APPROVED** [review]

**Comment by ncesario-lunarg** [comment]
Looks like the mock ICD tests are failing, but I don't see any log details from github actions. @mohfai-arm are you able to run the tests on the mock ICD with locally (note this has [changed](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/tests/README.md#running-tests-on-mockicd-and-profiles-layer) somewhat recently)?

**Comment by juan-lunarg** [comment]
> Looks like the mock ICD tests are failing, but I don't see any log details from github actions. @mohfai-arm are you able to run the tests on the mock ICD with locally (note this has [changed](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/tests/README.md#running-tests-on-mockicd-and-profiles-layer) somewhat recently)?

CI is acting up. I'm re-running the failed jobs right now.

**Comment by mohfai-arm** [comment]
I will have a look tomorrow, not sure what changed from last commit. First commit was passing all checks, and latest change only updated  few lines, so wasn't expecting any failure.

**Comment by juan-lunarg** [comment]
Rebasing to get github actions running. They are currently stuck...

**Comment by juan-lunarg** [comment]
Thanks for the contribution @mohfai-arm glad we got this in time for the upcoming SDK release 👍

**Pull request merged**